code/modules/reagents/reagent_containers/borghydro.dm

```diff
@@ -70,8 +70,10 @@
 			return
 
 	if (M.can_inject(user, 1))
-		user << "<span class='notice'>You inject [M] with the injector.</span>"
-		M << "<span class='notice'>You feel a tiny prick!</span>"
+		visible_message("<span class='notice'>[user] starts to inject [M] with their hypospray!</span>")
+		if(do_after(user, 3 SECONDS, act_mob = M))

```

Comment:
I wonder, check the regular syringes (human ones). See if they use `do_after` or `do_mob`. There's a nuanced difference and it should mimic this.